Adjust keybindings for deletion in the project panel

mikayla-maki opened the pull request:
Release Notes:

- Improve compatibility keybindings (Atom, JetBrains, TextMate)

mikayla-maki committed:
Adjust keybindings for deletion in the project panel

notpeter commented on assets/keymaps/default-macos.json:
```diff
@@ -613,8 +613,12 @@
-      "cmd-backspace": ["project_panel::Delete", { "skip_prompt": false }],
-      "cmd-delete": ["project_panel::Delete", { "skip_prompt": false }],
+      "shift-backspace": ["project_panel::Delete", { "skip_prompt": false }],
+      "shift-delete": ["project_panel::Delete", { "skip_prompt": false }],
```
cmd+backspace/delete and cmd+shift+backspace/delete look right.

Is there a reason we're adding shift+backspace and shift+delete? I've never seen them

notpeter committed:
Improve project panel delete behavior. Matches standard behavior.

- MacOS: cmd-backspace (trash without prompt)
- MacOS: cmd-option-backspace (immediate delete with prompt)
- Linux: delete (trash without prompt)
- Linux: shift-delete (immediate delete with prompt)
- All: Require custom keybinds for dangerous behavior (e.g. delete without prompt)

notpeter committed:
Restore pre-existing non-MacOS behavior. Bare delete key moves to trash but prompts.

notpeter commented:
I've made a few changes, but I feel this matches my expectations:

- MacOS:
  - cmd+backspace == trash without prompt (match finder)
  - cmd+shift+backspace == delete with prompt (match finder)
  - delete == trash with prompt (match vscode)
  - backspace (no-op, previously trash with prompt)
- Linux: Match native file explorers
  - delete == delete with prompt
  - shift+delete == delete with prompt
  - backspace (no-op, previously trash with prompt)

I feel relatively strongly that a backspace key should not delete files. This is not the behavior on Windows, MacOS or Linux (that I am aware) nor in VSCode.  I think it's just too easy to press backspace when you thought your focus was in the editor and not the project panel.

I could be convinced that delete on linux should be `delete with prompt`.

notpeter committed:
Show prompt for delete in linux.

notpeter committed:
Improve bindings for jetbrains and textmate while we're at it.

SomeoneToIgnore commented on assets/keymaps/default-linux.json:
```diff
@@ -587,7 +587,7 @@
       "enter": "project_panel::Rename",
       "shift-delete": ["project_panel::Delete", { "skip_prompt": false }],
-      "delete": ["project_panel::Trash", { "skip_prompt": true }],
+      "delete": ["project_panel::Trash", { "skip_prompt": false }],
```
To note, this alone will remove all keybindings from mac menu entries.

notpeter replied:
> To note, this alone will remove all keybindings from mac menu entries.

I don't understand, please say more.

1. Isn't this `-linux.json`?
2. How does adding a single binding to scoped to ProjectPanel remove all keybindings for mac menu entries?

SomeoneToIgnore replied:
Sorry, I've actually missed that this fixes the issue with Linux keybindings, and my comment is also wrong as was about mac, please disregard it entirely.

1. Yes, sorry.
2. It's not adding, but altering the keybinding's value what happens in the PR. If that gets out of sync with the value, defined in the menu, the binding keys won't be displayed in the UI.

In this case they actually start to match, so all good:
https://github.com/zed-industries/zed/blob/1e0e1c7c3199c412b49cbed56c7ad4087fc5842e/crates/project_panel/src/project_panel.rs#L504

notpeter replied:
Thank you. This is helpful to know that the menu items and keybinds could be out of sync in the actions they perform.

notpeter committed:
Fix Atom keybinds for project panel.

notpeter committed:
Add duplicate shortcut for project panel on Macos.

notpeter committed:
Restore existing non-standard deletion keybinds.
